fs/dafsserver/dafsserver.cpp

```diff
@@ -1630,15 +1620,90 @@ class CRemoteMarkupReadActivity : public CRemoteStreamReadBaseActivity, implemen
         CFieldFetcher(const RtlRecord &_recInfo, IColumnProvider *_currentMatch) : recInfo(_recInfo), currentMatch(_currentMatch)
         {
             numInputFields = recInfo.getNumFields();
+
+            // JCSMORE - should this be done (optionally) when RtlRecord is created?
+            for (unsigned fieldNum=0; fieldNum<numInputFields; fieldNum++)
+            {
+                if (recInfo.queryType(fieldNum)->queryChildType())
+                {
+                    const char *xpath = recInfo.queryXPath(fieldNum);
+                    dbgassertex(xpath);
+                    const char *ptr = xpath;
+                    char *expandedXPathPtr = nullptr;
+                    while (true)
+                    {
+                        if (*ptr == xpathCompoundSeparatorChar)
+                        {
+                            if (!compoundXPaths)
+                            {
+                                compoundXPaths = new const char *[numInputFields];
+                                memset(compoundXPaths, 0, sizeof(const char *)*numInputFields);
+                            }
+
+                            char *&expandedXPath = (char *&)compoundXPaths[fieldNum];
+                            size32_t sz = strlen(xpath)+1;
+                            expandedXPath = new char[sz];
+                            expandedXPathPtr = expandedXPath;
+                            if (ptr == xpath) // if leading char, just skip
+                                ++ptr;
+                            else
+                            {
+                                size32_t len = ptr-xpath;
+                                memcpy(expandedXPath, xpath, len);
+                                expandedXPathPtr = expandedXPath + len;
+                                *expandedXPathPtr++ = '/';
+                                ++ptr;
+                            }
+                            while (*ptr)
+                            {
+                                if (*ptr == xpathCompoundSeparatorChar)
+                                {
+                                    *expandedXPathPtr++ = '/';
+                                    ++ptr;
+                                }
+                                else
+                                    *expandedXPathPtr++ = *ptr++;
+                            }
+                        }
+                        else
+                            ptr++;
+                        if ('\0' == *ptr)
+                        {
+                            if (expandedXPathPtr)
+                                *expandedXPathPtr = '\0';
+                            break;
+                        }
+                    }
+                }
+            }
+        }
+        ~CFieldFetcher()
+        {
+            if (compoundXPaths)
+            {
+                for (unsigned fieldNum=0; fieldNum<numInputFields; fieldNum++)
+                    free((char *) compoundXPaths[fieldNum]);
```

Comment:
need to use delete [] if allocated with new []

Reply:
will fix.